### PR TITLE
Handle cassandra failures and add actor supervision

### PR DESCRIPTION
**Pull request by jypma**
@domask Care to give a preliminary review?

### CODE REVIEW

**Comment by jypma** [comment]
@domask ready for full review

File: src/main/scala/akka/persistence/cassandra/ResultSetActorPublisher.scala [hunk 1]
```diff
@@ -26,20 +32,29 @@ object ResultSetActorPublisher {
...
     case ResultSetReady(resultSet) =>
+      timeoutSchedule.cancel()
       log.debug("Resultset is ready with {} available results, exhausted: {}",
           resultSet.getAvailableWithoutFetching, resultSet.isExhausted)
       context become ready(resultSet)
       deliver(resultSet)
       unstashAll()
+      
+    case ExecTimeout =>
+      resultSetFuture.cancel(true)
```
Comment (domask) [inline src/main/scala/akka/persistence/cassandra/ResultSetActorPublisher.scala]: consider using named parameter, so it's more clear what boolean value stands for.

Comment (jypma) [inline src/main/scala/akka/persistence/cassandra/ResultSetActorPublisher.scala]: I would, but that's a Java API (cassandra client)

**Comment by domask** [comment]
:+1:

**Pull request merged**